### PR TITLE
remove deprecated analyzer api usage

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

**Comment by jakemac53** [comment]
> Do we need to bump the dep?

I decided not to since its only used in tests. We could, but we can't distinguish the version between regular deps and dev deps I don't think?

File: build_resolvers/test/resolver_test.dart [hunk 1]
```diff
@@ -367,8 +367,10 @@ int? get x => 1;
                 ''',
               }, (resolver) async {
                 var lib = await resolver.libraryFor(entryPoint);
-                expect(lib.languageVersionMajor, sdkLanguageVersion.major);
-                expect(lib.languageVersionMinor, sdkLanguageVersion.minor);
+                expect(lib.languageVersion.effective.major,
+                    sdkLanguageVersion.major);
```
Comment (kevmoo) [inline build_resolvers/test/resolver_test.dart:371]: nit: I'm starting to add trailing commas when I force a wrap in an arg list – do what you like, thoug!

Comment (jakemac53) [reply build_resolvers/test/resolver_test.dart:371]: we don't do that, generally

**Comment by jakemac53** [comment]
merged for now, this isn't published so we could still increase the dep in a followup if we think that is necessary

**Pull request merged**